test/cppsuite/src/main/thread_worker.cpp

```diff
@@ -210,6 +210,47 @@ thread_worker::remove(scoped_cursor &cursor, uint64_t collection_id, const std::
     return (ret == 0);
 }
 
+bool
+thread_worker::truncate(uint64_t collection_id, std::optional<std::string> start_key,
+  std::optional<std::string> stop_key, const std::string &config)
+{
+    WT_DECL_RET;
+
+    wt_timestamp_t ts = tsm->get_next_ts();
+    ret = txn.set_commit_timestamp(ts);
+    testutil_assert(ret == 0 || ret == EINVAL);
+    if (ret != 0) {
+        txn.set_needs_rollback(true);
+        return (false);
+    }
+
+    const std::string coll_name = database::build_collection_name(collection_id).c_str();
+
+    scoped_cursor start_cursor = session.open_scoped_cursor(coll_name);
+    scoped_cursor stop_cursor = session.open_scoped_cursor(coll_name);
+    if (start_key.has_value())
```

Comment:
You shouldn't need to call `.has_value()` std::optional is contextually converted to a boolean.

Reply:
That said it does help for readability in later lines.

Reply:
@luke-pearson, we should be cautious with this and we might add it to our guidelines.
Because it is an optional, `if (my_optional)` is indeed the same as `if (my_optional.has_value())`.

If the `my_optional` is a boolean and is `false`, `if (my_optional)` is still `true`.
This is the same for an optional that holds a pointer. Even if the pointer is `nullptr`, `if (my_optional)` is still `true`.
In fact, as long as `my_optional` contains a value, `if (my_optional)` returns `true` which can be misleading.

Reply:
I see where you're coming from but I think we should err on the side of not using .has_value(). Generally basic types like pointers and booleans shouldn't be contained within std::optionals.

Reply:
I'm happy with this, that's probably what we should update our guidelines with.